soda_machine.rb

```diff
@@ -1,18 +1,33 @@
 class SodaMachine
-  attr_reader :sodas, :cash
+  attr_reader :sodas, :cash, :sold_soda
 
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
+    @sold_soda = []
   end
 
   def current_inventory_count
+      @sodas.length
   end
 
   def find_soda(soda_brand)
+    if @sodas[:sodas].include? soda_brand
+      @sodas[:sodas][soda_brand]
+    else
+      nil
+    end
   end
 
   def sell(soda_brand)
+    if @sodas.include? soda_brand
```

Comment:
This seems reasonable until you look at what's in the `@sodas` array.  You're looking for a string in an array of `Soda` objects!  This will never come back true.